Assets/MixedRealityToolkit/_Core/Interfaces/InputSystem/IMixedRealityInputSource.cs

```diff
@@ -1,32 +1,78 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-using System.Collections;
 using Microsoft.MixedReality.Toolkit.Internal.Definitions;
+using System.Collections;
+using System.Collections.Generic;
 
 namespace Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem
 {
     /// <summary>
-    /// Interface for an input source.
-    /// An input source can be any user defined action that generally comes from a physical controller, sensor, or device.
+    /// Interface for a Mixed Reality Toolkit input source.
+    /// An input source can be any user defined input that generally comes from a physical controller, sensor, or device.
     /// </summary>
     public interface IMixedRealityInputSource : IEqualityComparer
     {
+        /// <summary>
+        /// The ID assigned to the Input Source
+        /// </summary>
         uint SourceId { get; }
 
+        /// <summary>
+        /// The Name assigned to the Input Source
+        /// </summary>
         string SourceName { get; }
 
+        /// <summary>
+        /// Outputs the current state of the Input Source, whether it is tracked or not. As defined by the SDK / Unity.
+        /// </summary>
+        InputSourceState InputSourceState { get; }
+
+        /// <summary>
+        /// The designated hand that the Input Source is managing, as defined by the SDK / Unity.
+        /// </summary>
+        Handedness Handedness { get; }
+
+        /// <summary>
+        /// The list of Pointers attached to the Input Source
+        /// </summary>
         IMixedRealityPointer[] Pointers { get; }
 
         /// <summary>
-        /// Returns the input info that the input source can provide.
+        /// Details the list of available buttons / interactions available from the Input Source.
         /// </summary>
-        InputType[] Capabilities { get; }
+        Dictionary<InputType,InteractionDefinition> Interactions { get; }
 
+        // TODO - Need to generalize the Input Type?
         /// <summary>
-        /// Returns whether the input source supports the specified input types.
+        /// Setup function to define controller
+        /// </summary>
+        /// <param name="interactionSourceState">SDK data to inform construction</param>
+        void SetupInputSource(UnityEngine.XR.WSA.Input.InteractionSourceState interactionSourceState);
+
+        // TODO - Need to generalize the Input Type?
+        /// <summary>
+        /// Update function to update controller
+        /// </summary>
+        /// <param name="interactionSourceState">SDK data to inform update</param>
+        void UpdateInputSource(UnityEngine.XR.WSA.Input.InteractionSourceState interactionSourceState);
```

Comment:
We shouldn't need to call update on input sources.  I'd rather let devs get the raw data at the time they need it instead of polling on updates.

Reply:
It means the controller does the lifting of translating raw input in to its data classes. That would be static for each controller and shouldn't really be touched (it's particular to that controller)

The Dev should be more interested with what they then do with that data.